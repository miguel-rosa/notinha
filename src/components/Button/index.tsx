import React, { FC } from 'react';
import { TouchableOpacity, StyleSheet, TextStyle} from 'react-native';
import Text from '../Text';

type ButtonProps = {
  onPress?():void;
  type?: 'primary' | 'secondary' | 'danger',
  size?: 'big' | 'medium' | 'small',
  useSpacing?: boolean;
  buttonStyle?: TextStyle;
}

const Button:FC<ButtonProps> = ({
  children,
  onPress = () => {},
  type = 'primary',
  size = 'medium',
  useSpacing = true,
  buttonStyle,
}) => {
  return (
    <TouchableOpacity accessibilityRole="button" onPress={onPress}
      style={[
        styles[`${type}Background`],
        styles[`${size}Button`],
        styles.button,
        {
          marginVertical: useSpacing ? 6 : 0,
        },
        buttonStyle,
        ]}>
      <Text
        style={[styles[`${type}Text`], styles[`${size}Text`], styles.text]}
        >{children}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    borderRadius: 8,
  },
  text: {

  },

  primaryBackground: {
    backgroundColor:'#0d66ff',
  },
  secondaryBackground: {
    backgroundColor:'transparent',
  },
  dangerBackground: {
    backgroundColor: 'rgb(255,88,88)',
  },
  bigButton: {
    paddingHorizontal: 20,
    paddingVertical: 16,
  },
  mediumButton: {
    paddingHorizontal: 16,
    paddingVertical: 12,
  },
  smallButton: {
    paddingHorizontal: 12,
    paddingVertical: 8,
  },
  bigText: {
    fontSize: 18,
  },
  mediumText: {
    fontSize: 16,
  },
  smallText: {
    fontSize: 14,
  },
  primaryText: {
    color: '#fff',
  },
  secondaryText: {
    color: '#040404',
  },
  dangerText: {
    color: '#fff',
  },
});

export default Button;
