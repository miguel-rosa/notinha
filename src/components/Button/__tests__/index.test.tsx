import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '..';

const innerText = 'Inner button text';

describe('given the prop onPress, it will be applied to the TouchableOpacity', () => {
  it('simple event handling', () => {
    let itPassed = false;

    const changeItPassed = () => itPassed = true;

    const { getByText } = render(<Button onPress={changeItPassed}>{innerText}</Button>);

    fireEvent.press(getByText(innerText));

    expect(itPassed).toBe(true);
  });
});

describe('CSS API', () => {
  it('type primary', () => {
    const tree = render(<Button type="primary">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('type secondary', () => {
    const tree = render(<Button type="secondary">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('type danger', () => {
    const tree = render(<Button type="danger">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('size big', () => {
    const tree = render(<Button size="big">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('size medium', () => {
    const tree = render(<Button size="medium">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('size small', () => {
    const tree = render(<Button size="small">{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('useSpacing true', () => {
    const tree = render(<Button useSpacing={true}>{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('useSpacing false', () => {
    const tree = render(<Button useSpacing={false}>{innerText}</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
