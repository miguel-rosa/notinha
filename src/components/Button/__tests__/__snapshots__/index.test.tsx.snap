// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS API size big 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 18,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API size medium 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API size small 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 14,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API type danger 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255,88,88)",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API type primary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API type secondary 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#040404",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API useSpacing false 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 0,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;

exports[`CSS API useSpacing true 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0d66ff",
      "borderRadius": 8,
      "marginVertical": 6,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Poppins_500Medium",
          "fontSize": undefined,
        },
        Array [
          Object {
            "color": "#fff",
          },
          Object {
            "fontSize": 16,
          },
          Object {},
        ],
      ]
    }
  >
    Inner button text
  </Text>
</View>
`;
